manubot/cite/isbn.py

```diff
@@ -0,0 +1,84 @@
+import collections
+import json
+import logging
+import re
+
+
+def get_isbn_citeproc(isbn):
+    return get_isbn_citeproc_citoid(isbn)
+
+
+def get_isbn_citeproc_isbnlib(isbn):
+    """
+    Generate CSL JSON Data for an ISBN using isbnlib.
+    """
+    import isbnlib
+    metadata = isbnlib.meta(isbn, cache=None)
+    csl_json = isbnlib.registry.bibformatters['csl'](metadata)
+    csl_data = json.loads(csl_json)
+    return csl_data
+
+
+def get_isbn_citeproc_citoid(isbn):
+    """
+    Return CSL JSON Data for an ISBN using the Wikipedia Citoid API.
+    https://en.wikipedia.org/api/rest_v1/#!/Citation/getCitation
+    """
+    import isbnlib
+    import requests
+    isbn = isbnlib.to_isbn13(isbn)
+    url = f'https://en.wikipedia.org/api/rest_v1/data/citation/mediawiki/{isbn}'
+    response = requests.get(url)
+    result = response.json()
+    if isinstance(result, dict):
```

Comment:
What type of object is `result` here if it is not a `dict`?

Reply:
If things go well, result is a list of one element.